### PR TITLE
[#1023, #875,  #1021, #1119] Fix loading scene / date from params, save / load view

### LINKED ISSUE #875: [FIX] Marking the selected date on the calendar and hour displayed on timeline
Tested on instance https://sok.grid.cyfronet.pl
 SOK version v14.1.0

**Steps:**
1. Go to /map/products
2. Click on the product Airmass 
3. Click on the link "przejdź do ostatniej zarejestrowanej sceny" (currently is 2020-07-07)

Color on the calendar.

-  Yellow - image is availlabe

- Red - date is selected and image is availlabe on this day (image should be load on the map)

- Blue - data is selected but image is not availlabe on this day.

- Gray - hover on the date

**What happens?**
the date is marked in gray, and other day is wrongly marked on the blue
obraz

**What was expected to happen?**
The date is marked in red.
obraz

------
**Steps:**
1. Change date from Airmass 2020-07-07 using arrows obraz to date 2020-07-03

**What happens?**
image

**What was expected to happen?**
image

--------

**Steps:**
1. Change product from previous step ("Airmass" 2020-07-03, where the marked on red is date 2020-07-07) to "Zachmurzenie (108m)"

**What happens?**

obraz

**What was expected to happen?**
obraz

-------

**Steps:**

1. Change product to "Chmury konwekcyjne" and chose the date from open calendar 2020-07-01 (date is marked correctly on red)
2. Click on the link "Zapisz konfigurację widoku" and confirm
3. Change product to "Chmury konwekcyjne PL" and chose the date from open calendar 2020-07-08 (date is marked correctly on red)
4. Click on the link "Wczytaj zapisana konfiguracje" and chose configuration from step 2.

**What happens?**
image

**What was expected to happen?**
image

### LINKED ISSUE #1021: Sentinel search form validation improvement nad handling 404 error for downloading non-existent data
- date input should have Validators.pattern(/[0-9]{4}-[0-9]{1,2}-[0-9]{1,2}/)
- number/float (cloudCover) inputs should have Validators.pattern(/([0-9]+,)?[0-9]+/)
- start date should be send to back-end with 00:00:00 time (convert example 2020-10-02 ->  2020-10-02T00:00:00)
- end date should be send to back-end with 23:59:59 time (convert example 2020-10-02 ->  2020-10-02T23:59:00)

### LINKED ISSUE #1023: [Fix] Save the image configuration
Tested on instance SOK version master-5a48c8f1715451ad89dd6a19842762528145620f

Steps:
1. Log as zkAdmin
2. Chose product Airmass 2020-04-01
3. Click on the link "Zapisz konfiguracje widoku w aplikacji" 
4. Click on the link "Wczytaj zapisana konfiguracje" and chose Airmass 2020-04-01

What happens?
last available scene is loaded (in this situation 2020-07-07)
image

What was expected to happen?
should be loaded scene from 2020-04-01

### LINKED ISSUE #1119: [Fix] Send the view to an email address
**Tested on instance SOK version**
master-2f465dba26a02ba303a4ee33d3503433c623d835

**Steps:**
1. Log as zkAdmin
2. Choose product Airmass
3. Click the link "Wyślij widok na adres email"
4. Fill in the form and send view
5. Click on received link

**What happens?**
the sent view is not loading, after click the link, the url doesnt have parameter &product=

**What was expected to happen?**
the sent view should load correctly

It is work locally

localhost: /map/products?product=11&scene=852&date=2020-02-02&zoom=6&centerx=2115070.325072198&centery=6800125.454397307

master:map/products?date=2020-02-02&zoom=6&centerx=2152102.1989128357&centery=6784204.8587500695

### PR DESCRIPTION
**Pull request by michal-szostak**
## What is in this PR?

* Fix loading configuration
* Fix loading scene from copied & pasted URL
* Add checkbox `include selected scene` in save configuration view
* Handle missing / nonexistent sentinel search download files (disable download link in list and in modal)

Closes #1023, closes #875, closes  #1021, closes #1119

### CODE REVIEW

**Comment by Marcelinna** [comment]
when saving the configuration without current date select, in the modal field date should be empty.
image

default name which is current date is misleading, field should be empty.

File: s4e-web/src/app/views/map-view/sentinel-search/search-results/search-results.component.html [hunk 1]
```diff
@@ -86,9 +86,11 @@
                   (click)="showDetails.emit(result)"
                 ></button>
                 <a
-                  [attr.href]="result.url"
+                  [attr.href]="result.url ? result.url : 'javascript:undefined'"
                   (click)="isUserLoggedIn || interceptDownload($event)"
-                  target="_blank"
+                  [attr.target]="result.url ? '_blank' : ''"
+                  [attr.title]="result.url ? '' : 'Ten produkt nie posiada pliku'"
```
Comment (jswk) [inline s4e-web/src/app/views/map-view/sentinel-search/search-results/search-results.component.html:92]: Rather "Ta scena nie posiada artefaktu zip", as the scene will contain at least a single file anyway (metadata file).
Also, in the details modal we use the "artefakt" nomenclature and I would stick to it.

File: s4e-web/src/app/views/map-view/state/map/map.service.ts [hunk 2]
```diff
@@ -121,20 +122,28 @@ export class MapService {
...
+      this.store.setLoading(true);
+      this.productService.setSelectedDate(viewConfig.date);
+      this.productService.setManualDate(viewConfig.manualDate);
+      this.setView(viewConfig.viewPosition);
+      this.overlayService.setAllActive(viewConfig.overlays);
+    });
+
+    return this.productService.setActive$(viewConfig.productId).pipe(
+      tap(() =>
+        console.log('updateStoreByView$ - SCENE', viewConfig.sceneId, viewConfig)
```
Comment (jswk) [inline s4e-web/src/app/views/map-view/state/map/map.service.ts:142]: I'd remove that

File: s4e-web/src/app/views/map-view/state/product/product.service.ts [hunk 3]
```diff
@@ -198,6 +200,7 @@ export class ProductService {
 
         return !!data.sceneId;
       }),
+      logIt('getLastAvailableScene$.setSelectedDate'),
```
Comment (jswk) [inline s4e-web/src/app/views/map-view/state/product/product.service.ts:203]: It's also rather unnecessary log, but there are several such invocations throughout the app, so I'm not 100% sure about this:
image

**Comment by Marcelinna** [comment]
Date format is incorrect, field date is fill for uncheck checkbox too.
image
image

**Review by Marcelinna — APPROVED** [review]

**Review by jswk — APPROVED** [review]

**Pull request merged**